libs/search/src/lib/elasticsearch/elasticsearch.model.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  DefaultOperator,
+  ExpandWildcards,
+  GenericParams,
+  NameList,
+  TimeSpan,
+} from 'elasticsearch'
+
+export type SortOrder = 'asc' | 'desc'
+interface SortParam {
+  [key: string]: SortOrder
+}
+export type SortParams = string | SortParam | (string | SortParam)[]
+
+export interface SearchParams extends GenericParams {
+  analyzer?: string
```

Comment:
missing `aggs` ?
I think the `SearchParams` interface need a complete refactoring, where this change could be part of.
Old interface might still be used somewhere, some object might not be correctly typed (otherwise `aggs` would trigger an error), es-types dep still in the project etc..

Reply:
> missing `aggs` ?

the type comes from the `@types/elasticsearch` https://github.com/geonetwork/geonetwork-ui/commit/5bfae11199fe1f3f49515db2cdadf779f210cb91

Reply:
Why closing ? Just remove the `SearchParams` interface from the PR and it's great !

Reply:
> Why closing ?

Un peu fatigué.



> Just remove the `SearchParams` interface from the PR and it's great !

If I remove it then it does not build. If we want buildPayload to return a typed response we have to create a SearchParams which is not an Elasticsearch 5 version type.